site2/docs/security-extending.md

```diff
@@ -52,7 +52,19 @@ authenticationProviders=
 
 ```
 
-For the implementation of the `org.apache.pulsar.broker.authentication.AuthenticationProvider` interface, refer to [here](https://github.com/apache/pulsar/blob/master/pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authentication/AuthenticationProvider.java).
+:::tip
+
+Pulsar supports an authentication provider chain that contains multiple authentication providers with the same authentication method name. 
+
+For example, your Pulsar cluster uses JSON Web Token (JWT) authentication and you want to upgrade it to use OAuth2.0 authentication. Both JWT and OAuth2.0 share the same authentication method name. In this case, you can chain the two class names in `authenticationProviders` and separate them by using a comma.
+
+```properties
+authenticationProviders=org.apache.pulsar.broker.authentication.AuthenticationProviderJWT,org.apache.pulsar.broker.authentication.AuthenticationProviderOAuth2
```

Comment:
There is no `AuthenticationProviderJWT`, instead, it should be `AuthenticationProviderToken`.

There is no `AuthenticationProviderOAuth2` in Pulsar repo as well. Pulsar uses `AuthenticationProviderToken`  to enable OAuth2 authentication as well, see https://pulsar.apache.org/docs/next/security-oauth2#broker-configuration.

https://github.com/apache/pulsar/pull/9094 is not a PR that supports chained authentication providers, it supports chained authentication providers **with the same auth method name**.

Reply:
The chained authentication providers means multiple authentication providers can be configured at the same time. For example, with the following config

```properties
authenticationProviders=org.apache.pulsar.broker.authentication.AuthenticationProviderTls,org.apache.pulsar.broker.authentication.AuthenticationProviderToken
```

Pulsar can enable authentication for [JWT](https://pulsar.apache.org/docs/next/security-jwt/#enable-token-authentication-on-brokers) and [TLS](https://pulsar.apache.org/docs/next/security-tls-authentication#enable-tls-authentication-on-brokers) at the same time.

Reply:
The auth method name of `AuthenticationProviderTls` is "tls" and the auth method name of `AuthenticationProviderToken` is "token".

The motivation of #9094 is that since we can implement our own `AuthenticationProvider`, for example, we can implements `AuthenticationProviderOAuth2` whose auth name is also "token" in future, at that time we can configure both `AuthenticationProviderOAuth2` and `AuthenticationProviderToken`.

Reply:
> The chained authentication providers means multiple authentication providers can be configured at the same time. For example, with the following config
> 
> ```ini
> authenticationProviders=org.apache.pulsar.broker.authentication.AuthenticationProviderTls,org.apache.pulsar.broker.authentication.AuthenticationProviderToken
> ```
> 
> Pulsar can enable authentication for [JWT](https://pulsar.apache.org/docs/next/security-jwt/#enable-token-authentication-on-brokers) and [TLS](https://pulsar.apache.org/docs/next/security-tls-authentication#enable-tls-authentication-on-brokers) at the same time.

The old version can also do this thing. 

The chained authentication version can support you configure the multiple authentication providers that has same auth name.

Reply:
> The motivation of #9094 is that since we can implement our own `AuthenticationProvider`, for example, we can implements `AuthenticationProviderOAuth2` whose auth name is also "token" in future, at that time we can configure both `AuthenticationProviderOAuth2` and `AuthenticationProviderToken`.


This is right.

Reply:
Updated. PTAL.